GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -3335,9 +3335,31 @@ private void SubmoduleToolStripButtonClick(object sender, EventArgs e)
             }
         }
 
+        private void PreventToolStripSplitButtonClosing(ToolStripSplitButton control)
+        {
+            if (toolStripBranchFilterComboBox.Focused == false && toolStripRevisionFilterTextBox.Focused == false)
+            {
+                control.Tag = this.FindFocusedControl();
+                control.DropDownClosed += ToolStripSplitButtonDropDownClosed;
+                toolStripBranchFilterComboBox.Focus();
+            }
+        }
+
+        private void ToolStripSplitButtonDropDownClosed(object sender, EventArgs e)
+        {
+            var control = sender as ToolStripSplitButton;
+            control.DropDownClosed -= ToolStripSplitButtonDropDownClosed;
+            if (control.Tag != null)
+            {
+                ((Control)control.Tag).Focus();
+                control.Tag = null;
+            }
+        }
+
         private void toolStripButtonLevelUp_DropDownOpening(object sender, EventArgs e)
         {
             LoadSubmodulesIntoDropDownMenu();
+            PreventToolStripSplitButtonClosing((ToolStripSplitButton)sender);
```

Comment:
here and other places, why do unsafe cast when you know exactly which toolstrip button should be actioned?

```
private void toolStripButtonLevelUp_DropDownOpening(object sender, EventArgs e)
{
    LoadSubmodulesIntoDropDownMenu();
    PreventToolStripSplitButtonClosing(toolStripButtonLevelUp);
}
```

The reason being - too often in the codebase handler get invoked prgrammatically, and not always with the necessary parameters.